pkg/sshd/circuit.go

```diff
@@ -0,0 +1,61 @@
+package sshd
+
+import (
+	"fmt"
+	"sync/atomic"
+)
+
+// CircuitState represents the state of a Circuit
+type CircuitState uint32
```

Comment:
> Now that I've gotten a few questions on it, I wonder if it'd be best to refactor it using a mutex, which seems to be more familiar. Thoughts?

Honestly it's all bikeshedding, which I *hate*. I'm quite happy with this implementation, and it serves its purpose as required, so it looks good to me. :)

On the plus side, I just learned about a new package today! :tada: